### PR TITLE
Avoid unnecessary process creation

### PR DESCRIPTION
**Pull request by dnfield**
This stops leaking flutter_tester.exe from expression_evaluation_test.  We were launching a flutter_tester device that we don't need and we dont' try to kill.  Instead, just run the tests in `--start-paused` state.

Part of https://github.com/flutter/flutter/issues/51421 - unfortunately, we're still leaking some folders in this case.

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
RSLGTM

File: packages/flutter_tools/test/integration.shard/test_driver.dart [hunk 1]
```diff
@@ -679,8 +679,7 @@ class FlutterTestTestDriver extends FlutterTestDriver {
       'test',
       '--disable-service-auth-codes',
       '--machine',
-      '-d',
-      'flutter-tester',
+      '--start-paused',
```
Comment (aam) [inline packages/flutter_tools/test/integration.shard/test_driver.dart]: We seem to be adding '--start-paused' argument already at line 89 conditional on withDebugger=true(which is the case for expression evaluation tests).

File: packages/flutter_tools/test/integration.shard/test_driver.dart [hunk 2]
```diff
@@ -679,8 +679,7 @@ class FlutterTestTestDriver extends FlutterTestDriver {
       'test',
       '--disable-service-auth-codes',
       '--machine',
-      '-d',
-      'flutter-tester',
```
Comment (aam) [inline packages/flutter_tools/test/integration.shard/test_driver.dart]: Conceptually we are still launching flutter_tester for these tests, right? Who is going to terminate it?

Comment (dnfield) [reply packages/flutter_tools/test/integration.shard/test_driver.dart]: Previously we were launching two.  Now we launch one.  Testing locally on Windows, this doesn't leak the flutter_tester.exe process we used to leak.

File: packages/flutter_tools/test/integration.shard/test_driver.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter_tools/test/integration.shard/test_driver.dart]: Ahh thanks, I'll remove this then

**Comment by dnfield** [comment]
This flag has no effect on `flutter test` AFAICT - I just got lucky that it iddn't leak on the particular run I tried.

**Pull request closed without merging**